src/parser.cpp

```diff
@@ -364,7 +364,7 @@ namespace Sass {
   Definition_Obj Parser::parse_definition(Definition::Type which_type)
   {
     Scope parent = stack.empty() ? Scope::Rules : stack.back();
-    if (parent != Scope::Root && parent != Scope::Rules && parent != Scope::Function) {
+    if (parent != Scope::Root && parent != Scope::Rules && parent != Scope::Function && parent != Scope::Media) {
```

Comment:
This belongs in the check nesting visitor for now. The parser should avoid validating the AST tree.